capstone/src/main/java/com/google/sps/servlets/EventListing.java

```diff
@@ -3,50 +3,46 @@
 import com.google.api.client.util.DateTime;
 import com.google.api.services.calendar.Calendar;
 import com.google.api.services.calendar.model.Event;
-import com.google.api.services.calendar.model.EventAttendee;
 import com.google.api.services.calendar.model.EventDateTime;
+import com.google.sps.data.VotingEvent;
 import java.io.IOException;
-import java.util.Arrays;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 @WebServlet("/eventlisting")
 public class EventListing extends HttpServlet {
-  
+  VotingEvent votingEvent;
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    votingEvent = (VotingEvent) request.getAttribute("event");
```

Comment:
Will doPost always be called before doGet?
My concern is that otherwise, this won't be set.

Reply:
So the doPost will only be called when the user clicks to add to calendar button, the doGet will always be called first because the homepage has to redirect to eventlisting first (which triggers the doGet) therefore it shouldn't be possible for the user to trigger the doPost first (unless they change the url directly to eventlisting.html and then click the button bypassing the homepage)

Reply:
Ah shoot had the two reversed in my initial messages. Sounds good though.
I think to make this a bit more robust, you could just add an if(votingEvent == null) {} check to the doPost

Reply:
Ah okay, makes sense, smart!

Reply:
I did try this, but then it wouldn't build so I might just include it with the next pull request I make after testing it in cloud shell a bit.